test/system_settings_test.rb

```diff
@@ -0,0 +1,361 @@
+    end
+
+    context "is not in normal mode" do
+      before do
+        allow(Yast::Mode).to receive(:mode).and_return("installation")
+      end
+
+      it "does not read bootloader configuration" do
+        expect(Yast::Bootloader).to_not receive(:Read)
+        settings.Read
+      end
+
+    end
+  end
+
+  describe "#Activate" do
+    let(:sysrq_keys) { false }
+    let(:scheduler)  { "" }
+    let(:disk)       { "/sys/block/sda/queue/scheduler" }
+
+    before do
+      settings.SetSysRqKeysEnabled(sysrq_keys)
+      settings.SetIOScheduler(scheduler)
+      allow(Yast::Bootloader).to receive(:modify_kernel_params)
+      allow(Yast::Bootloader).to receive(:proposed_cfg_changed=)
+      allow(Dir).to receive(:[]).with(/scheduler/).and_return([disk])
+    end
+
+    context "given that SysRq keys status is unknown" do
+      it "does not update /proc/sys/kernel/sysrq" do
+        settings.main
+        expect(Yast::SCR).to_not receive(:Execute)
+          .with(Yast::Path.new(".target.bash"), /sysrq/)
+        settings.Activate
+      end
+    end
+
+    context "given that SysRq keys are enabled" do
+      let(:sysrq_keys) { true }
+
+      it "writes '1' to /proc/sys/kernel/sysrq" do
+        expect(Yast::SCR).to receive(:Execute)
+          .with(Yast::Path.new(".target.bash"), "echo '1' > /proc/sys/kernel/sysrq")
+        settings.Activate
+      end
+    end
+
+    context "given that SysRq keys is disabled" do
+      it "writes '0' to /proc/sys/kernel/sysrq" do
+        expect(Yast::SCR).to receive(:Execute)
+          .with(Yast::Path.new(".target.bash"), "echo '0' > /proc/sys/kernel/sysrq")
+        settings.Activate
+      end
+    end
+
+    context "given that a scheduler is set" do
+      let(:scheduler) { "cfq" }
+
+      it "updates bootloader configuration" do
+        expect(Yast::Bootloader).to receive(:modify_kernel_params)
+          .with("elevator" => scheduler)
+        expect(Yast::Bootloader).to receive(:proposed_cfg_changed=).with(true)
+        allow(File).to receive(:write)
+        settings.Activate
+      end
+
+      it "activates scheduler for all disk devices" do
+        expect(File).to receive(:write).with(disk, scheduler)
+        settings.Activate
+      end
+    end
+
+    context "given that no scheduler is set" do
+      let(:scheduler) { "" }
+
+      it "removes parameter from bootloader configuration" do
+        expect(Yast::Bootloader).to receive(:modify_kernel_params)
+          .with("elevator" => :missing)
+        expect(Yast::Bootloader).to receive(:proposed_cfg_changed=).with(true)
+        settings.Activate
+      end
+
+      it "does not activate scheduler" do
+        expect(File).to_not receive(:write)
+        settings.Activate
+      end
+    end
+  end
+
+  describe "#Write" do
+    let(:sysctl_sysrq) { "0" }
+    let(:mode) { "normal" }
+
+    before do
+      allow(Yast::Mode).to receive(:mode).and_return(mode)
+      allow(Yast::SCR).to receive(:Read).and_call_original
```

Comment:
NP: Move this mock below, the standalone line does not make sense, it only makes sense with the other `allow(Yast::SCR)` below. So group them together.

(My original comment was that this line is useless, then I noticed the other mock later.)